### PR TITLE
Make TargetPlatformVariant.all the default for flutter widget tests

### LINKED ISSUE #86817: Make TargetPlatformVariant.all() the default for widget tests in flutter/flutter
Tracking issue for [flutter.dev/go/test-all-platforms](flutter.dev/go/test-all-platforms)

### PR DESCRIPTION
**Pull request by Piinks**
This changes the default platform variant for widgets tests from just `TargetPlatform.android` to `TargetPlatformVariant.all()`.
The motivation for this change is described in the design document at [flutter.dev/go/test-all-platforms](flutter.dev/go/test-all-platforms)

Fixes https://github.com/flutter/flutter/issues/86817

Will require follow up in https://github.com/flutter/flutter/issues/87682 for broken test files. 
These files are broken due to 
- leaked states or dependencies
- some tests are platform-specific and rely on the previous default
- many are affected by default scrollbars on desktop platforms as those tests require a MediaQuery to be present

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/flutter_test_config.dart [hunk 1]
```diff
@@ -19,6 +19,9 @@ Future<void> testExecutable(FutureOr<void> Function() testMain) {
   // receive the event.
   WidgetController.hitTestWarningShouldBeFatal = true;
 
+  // Run widget tests across all platforms by default.
+  defaultPlatformVariant = TargetPlatformVariant.all();
```
Comment (Piinks) [inline packages/flutter/test/flutter_test_config.dart:23]: Points of Interest: This is where the default is set

File: packages/flutter_test/lib/src/widget_tester.dart [hunk 2]
```diff
@@ -140,9 +140,10 @@ void testWidgets(
   test_package.Timeout? timeout,
   Duration? initialTimeout,
   bool semanticsEnabled = true,
-  TestVariant<Object?> variant = const DefaultTestVariant(),
+  TestVariant<Object?>? variant,
   dynamic tags,
 }) {
+  variant ??= (defaultPlatformVariant ?? const DefaultTestVariant()) as TestVariant<Object?>;
```
Comment (Piinks) [inline packages/flutter_test/lib/src/widget_tester.dart:146]: Another point fo interest, chnanges to testWidgets

File: packages/flutter_test/lib/src/widget_tester.dart [hunk 3]
```diff
@@ -252,6 +253,9 @@ class DefaultTestVariant extends TestVariant<void> {
   Future<void> tearDown(void value, void memento) async {}
 }
 
+///
+TargetPlatformVariant? defaultPlatformVariant;
```
Comment (Piinks) [inline packages/flutter_test/lib/src/widget_tester.dart:257]: And here. The rest of the changes are just a whole lot of opt-outs for the broken tests

**Comment by Piinks** [comment]
@goderbauer I have added comments to the points of interest above. There are a lot of tests to repair after this, but most of them are simple (i.e. most scrolling tests are missing media queries needed by desktop scrollbars)

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 10.
View them at https://flutter-gold.skia.org/cl/github/87680

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 11.
View them at https://flutter-gold.skia.org/cl/github/87680

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 12.
View them at https://flutter-gold.skia.org/cl/github/87680

**Comment by Piinks** [comment]
After discussing with other contributors, I do not think we are going to pursue this change right now, the newly introduced test randomization needs time to stabilize. This change would exacerbate the amount of potentially broken tests from day to day when the random seed changes. Will leave the issue open to revisit.

**Pull request closed without merging**